src/components/MapView/index.tsx

```diff
@@ -133,7 +132,7 @@ function MapView({ classes }: MapViewProps) {
 const styles = () =>
   createStyles({
     container: {
-      position: 'relative',
+      height: '100%',
```

Comment:
```suggestion
      height: '100%',
      position: 'relative',
```
I believe both issues will be solved by re-introducing `position: 'relative'` so both the loading screen and legend are positioned from the non-static map container instead of body.